src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunctionDefinition.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Threading.Tasks;
+
+#pragma warning disable S1694 // An abstract class should have both abstract and concrete methods
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents a function that can be described to an AI service.</summary>
+public abstract class AIFunctionDefinition : AITool
```

Comment:
Just thinking aloud - but would it be possible to make `AIFunctionDefinition` a concrete type (with the constructor parameters to instantiate the four virtual properties)?

Would someone want to inherit this abstract class for any reason other than providing custom values for these four properties (i.e., name, description and the two schemas)?